src/Icons/Models/IconLink.cs

```diff
@@ -0,0 +1,212 @@
+    public IAttr? Type { get; }
+    public IAttr? Sizes { get; }
+    public Uri ParentUri { get; }
+    public string BaseUrlPath { get; }
+    public int Priority
+    {
+        get
+        {
+            if (_width == null || _width != _height)
+            {
+                return 200;
+            }
+
+            return _width switch
+            {
+                32 => 1,
+                64 => 2,
+                >= 24 and <= 128 => 3,
+                16 => 4,
+                _ => 100,
+            };
+        }
+    }
+
+    public IconLink(Uri parentPage)
+    {
+        _useUriDirectly = true;
+        _validated = true;
+        ParentUri = parentPage;
+        BaseUrlPath = parentPage.PathAndQuery;
+    }
+
+    public IconLink(IElement element, Uri parentPage, string baseUrlPath)
+    {
+        Href = element.Attributes["href"];
+        ParentUri = parentPage;
+        BaseUrlPath = baseUrlPath;
+
+        Rel = element.Attributes["rel"];
+        Type = element.Attributes["type"];
+        Sizes = element.Attributes["sizes"];
+
+        if (!string.IsNullOrWhiteSpace(Sizes?.Value))
+        {
+            var sizeParts = Sizes.Value.Split('x');
+            if (sizeParts.Length == 2 && int.TryParse(sizeParts[0].Trim(), out var width) &&
+                int.TryParse(sizeParts[1].Trim(), out var height))
+            {
+                _width = width;
+                _height = height;
+            }
+        }
+    }
+
+    public bool IsUsable()
+    {
+        if (string.IsNullOrWhiteSpace(Href?.Value))
+        {
+            return false;
+        }
+
+        if (Rel != null && _iconRels.Contains(Rel.Value.ToLower()))
+        {
+            _validated = true;
+        }
+        if (Rel == null || !_blacklistedRels.Contains(Rel.Value.ToLower()))
+        {
+            try
+            {
+                var extension = Path.GetExtension(Href.Value);
+                if (_iconExtensions.Contains(extension.ToLower()))
+                {
+                    _validated = true;
+                }
+            }
+            catch (ArgumentException) { }
+        }
+        return _validated;
+    }
+
+    /// <summary>
+    /// Fetches the icon from the Href. Will always fail unless first validated with IsUsable().
+    /// </summary>
+    public async Task<Icon?> FetchAsync(ILogger<IIconFetchingService> logger, IHttpClientFactory httpClientFactory)
+    {
+        if (!_validated)
+        {
+            return null;
+        }
+
+        var uri = BuildUri();
+        if (uri == null)
+        {
+            return null;
+        }
+
+        using var response = await IconHttpRequest.FetchAsync(uri, logger, httpClientFactory);
+        if (!response.IsSuccessStatusCode)
+        {
+            return null;
+        }
+
+        var format = response.Content.Headers.ContentType?.MediaType;
+        var bytes = await response.Content.ReadAsByteArrayAsync();
+        response.Content.Dispose();
```

Comment:
I was surprised by this in the original code, too.

I read a few blogposts that suggested disposing content when you finish reading it, but aren't going to drop the response out of context for a while. In this example that probably doesn't make sense, since we're immediately returning.

If we run in to connection pool or memory issues we can look here, I guess.

Reply:
After @joseph-flinn 's comment that we have issues with Icon service memory management in prod already, I'd like to keep this in.

I realize it _should_ be cleaned, but calling dispose multiple times by definition does nothing bad.